server/routes/orion.js

```diff
@@ -32,231 +32,141 @@ tv4.addSchema('review', schema.review);
-  var timeframeQuery;
-  var actualOccupancyLevels;
-  var restaurantReservations;
   var occupancyLevelsObject;
-  var restaurant;
+  var servicePath;
   var fiwareHeaders;
-  var actualDate = new Date().getTime();
-  timeframeQuery = utils.getTimeframe(actualDate);
-  utils.sendRequest('GET', {
-        'type': 'FoodEstablishmentReservation',
-        'q': timeframeQuery,
-        'limit': 1000
-      },
-      null,
-      req.headers)
-    .then(function(data) {
-      restaurantReservations = data.body;
-      actualOccupancyLevels = utils.getOccupancyLevels(
-        restaurantReservations,
-        req.params.id);
-      console.log(actualOccupancyLevels);
-      occupancyLevelsObject = utils.updateOccupancyLevels(
-        actualOccupancyLevels, actualDate);
-      utils.getListByType('Restaurant', req.params.id, req.headers)
-        .then(function(data) {
-          restaurant = data.body;
-          // add service path for the restaurant
-          fiwareHeaders = JSON.parse(JSON.stringify(req.headers));
-          if (typeof restaurant.department !== 'undefined' &&
-            restaurant.department !== '') {
-            fiwareHeaders['fiware-servicepath'] = '/' + restaurant.department;
-          }
-          utils.sendRequest('PATCH',
-              occupancyLevelsObject,
-              req.params.id,
-              fiwareHeaders)
-            .then(function(data) {
-              utils.getListByType('Restaurant', req.params.id, req.headers)
-                .then(function(data) {
-                  res.statusCode = data.statusCode;
-                  res.json(utils.dataToSchema(data.body));
-                })
-                .catch(function(err) {
-                  res.statusCode = err.statusCode;
-                  res.json(err.error);
-                });
-            })
-            .catch(function(err) {
-              res.statusCode = err.statusCode;
-              res.json(err.error);
-            });
-        })
-        .catch(function(err) {
-          res.statusCode = err.statusCode;
-          res.json(err.error);
-        });
-    })
-    .catch(function(err) {
-      res.statusCode = err.statusCode;
-      res.end();
-    });
+  var restaurantId = utils.generateId(req.params.id);
+  var currentDate = new Date().toISOString();
+  var sqlList = [];
+  var queryString = {};
+  var timeframe = utils.getTimeframe(currentDate);
+  utils.addConditionToQuery(sqlList, 'startTime', '==', timeframe);
+  utils.addConditionToQuery(sqlList, 'reservationFor', '==', req.params.id);
+  utils.addConditionToQuery(sqlList, 'reservationStatus', '==', 'Confirmed');
+  queryString.q = sqlList.join(';');
+  utils.getListByType('FoodEstablishmentReservation', null, req.headers,
+      queryString)
+  .then(function(reservations) {
+    occupancyLevelsObject = utils.updateOccupancyLevels(reservations.body,
+      currentDate);
+    return utils.getListByType('Restaurant', restaurantId, req.headers);
```

Comment:
use a constant for the entity type 'Restaurant'. Same for the rest of entity types